sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1788,17 +1788,20 @@ class AdaptiveQueryExecSuite
 
   test("SPARK-35650: Coalesce number of partitions by AEQ") {
     withSQLConf(SQLConf.COALESCE_PARTITIONS_MIN_PARTITION_NUM.key -> "1") {
-      val query = "SELECT /*+ REPARTITION */ * FROM testData"
-      val (_, adaptivePlan) = runAdaptiveAndVerifyResult(query)
-      collect(adaptivePlan) {
-        case r: CustomShuffleReaderExec => r
-      } match {
-        case Seq(customShuffleReader) =>
-          assert(customShuffleReader.partitionSpecs.size === 1)
-          assert(!customShuffleReader.isLocalReader)
-        case _ =>
-          fail("There should be a CustomShuffleReaderExec")
-      }
+      Seq("REPARTITION", "REBALANCE_PARTITIONS", "REBALANCE_PARTITIONS(key)")
```

Comment:
can we exclude `REBALANCE_PARTITIONS`? I think we should be able to apply local shuffle reader for it, in the future.

Reply:
yea, removed it now